raiden_contracts/contracts/TokenNetwork.sol

```diff
@@ -281,35 +281,36 @@ contract TokenNetwork is Utils {
         isOpen(channel_identifier)
         public
     {
-        require(channel_identifier == getChannelIdentifier(participant, partner));
+        uint256 added_deposit;
+        uint256 channel_deposit;
+
+        // Check deposit constraints
         require(total_deposit > 0);
         require(total_deposit <= deposit_limit);
 
-        uint256 added_deposit;
-        uint256 channel_deposit;
+        // Check that address data is correct
+        require(channel_identifier == getChannelIdentifier(participant, partner));
 
+        // Get channel deposit state
         Channel storage channel = channels[channel_identifier];
         Participant storage participant_state = channel.participants[participant];
         Participant storage partner_state = channel.participants[partner];
 
-        // Calculate the actual amount of tokens that will be transferred
+        // Check for underflow:
+        require(total_deposit > participant_state.deposit);
```

Comment:
Checking for underflows before the actual state change is confusing in my opinion. 

As discussed in https://github.com/raiden-network/raiden-contracts/issues/47#issuecomment-413898659,  I think that a computation  should be `followed by the validation of the computation`.

> But I would strive for a contract-based approach, where the beginning of a function starts with the requirements, first validating/requiring the value inputs, then the code is followed by the computation, which is followed by the validation of the computation and external side-effects, and the code finishes with the asserts/invariants that must be true after the execution of the code.

Reply:
I followed the approach described in the comment, for me overflow checks should be part of 
> first validating/requiring the value inputs
as we must not perform any compuations on overflowing input values.

Reply:
> Validating input would be something like if say an integer should not have the value 0 or something like that.

I'd say that is pretty much what we do here. We validate, that `total_deposit` is monotonically increasing. Additionally to saving gas, it makes the code easier to read/audit to me, when we first formulate constraints on the inputs/variables, then directly do computation. However, we can not do this directly after the function is called, but only after reading from the state, so I see why you'd classify this as validation of computation.